---
title: "CITEVizTestData"
date: "`r Sys.Date()`"
author:
  - name: Garth Kong
    affiliation:
      "Knight Cancer Institute, Oregon Health and Science University"
  - name: Thai Nguyen
    affiliation:
      "Bioinformatics and Genomics Masters Program, University of Oregon"
  - name: Wesley Rosales
    affiliation:
      "Bioinformatics and Genomics Masters Program, University of Oregon"
output:
    BiocStyle::html_document:
      toc: true
vignette: >
  %\VignetteIndexEntry{'CITEVizTestData'}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r setup, include=FALSE, eval = FALSE}
knitr::opts_chunk$set(echo = TRUE)

BiocStyle::markdown()
```

# About CITEViz

CITEViz is an RShiny app that allows users to classify cell types in CITE-Seq data via cellular surface protein levels. Similar to flow cytometry, users can physically draw gates and filter on a mixture of cells, and the selected cells are immediately highlighted in dimensional reduction space (e.g. UMAP, tSNE, PCA). The main advantage of CITEViz is its interactive workflow, which provides rapid feedback to the users to help classify cell types.

Additional features of CITEViz includes assessment of quality control metrics, feature visualization (feature expression and co-expression), and plots of cell clusters (2D and optionally 3D). The input of CITEViz is a pre-processed Seurat/SingleCelExperiment object with assays containing normalized RNA/SCT (single-cell transform) and ADT (antibody-derived tag) count data, as well as data for at least one dimensional reduction (e.g. UMAP, PCA, etc.).

# Preparation of the Test Data

In this tutorial, we use a peripheral blood mononuclear cells (PBMC) CITE-Seq dataset which was generated from the [Satija Lab](https://satijalab.org/) by [Hao et. al. 2021](https://www.sciencedirect.com/science/article/pii/S0092867421005833?via%3Dihub) under the [Creative Commons BY 4.0](https://creativecommons.org/licenses/by/4.0/) license. The original dataset (160K cells x 228 antibodies) was down-sampled (2500 cells x 228 antibodies) for demonstration purposes, which are detailed in the `inst/scripts/make-data.R` file in the ExperimentHub.

# Download CITEViz Test Data from ExperimentHub

The following code chunk shows how to access the PBMC CITE-Seq dataset through ExperimentHub. The test data are identical but formatted in either a Seurat object or a SingleCellExperiment. When either of these datasets are downloaded, it is exported as an RDS for input to CITEViz via the "upload" button in the GUI. There are multiple ways to download from ExperimentHub, such as the following:

```{r eval = FALSE}
library(ExperimentHub)

# create EH instance and query it
eh <- ExperimentHub()
cv <- query(eh, "CITEVizTestData")

# get PBMC CITE-Seq datasets in two formats
file_1 <- cv[[1]] # Test data in Seurat format
file_2 <- cv[[2]] # Test data in SingleCellExperiment format
```

```{r eval = FALSE}
library(ExperimentHub)

# create EH instance and query it
eh <- ExperimentHub()
cv <- query(eh, "CITEVizTestData")

# another way to load the test data
file_1 <- CITEVizTestData(format = "Seurat")
file_2 <- CITEVizTestData(format = "SingleCellExperiment")
```

```{r eval = FALSE}
library(ExperimentHub)

# create EH instance and query it
eh <- ExperimentHub()
cv <- query(eh, "CITEVizTestData")

# use the following to import by EH ID
# file_1 <- CITEVizTestData(format = "Seurat")
# file_2 <- CITEVizTestData(format = "SingleCellExperiment")

# export information to external file
saveRDS(file_1, "CITEVizTestData_Seurat.rds")
saveRDS(file_2, "CITEVizTestData_SingleCellExperiment.rds")

```

After exporting both datasets, they are downloaded your local working directory. Both files can be the input to CITEViz. To understand how to use CITEViz, please click the link [here](https://github.com/maxsonBraunLab/CITEViz).

# References

Hao, Y. et al. Integrated analysis of multimodal single-cell data. Cell 184, 3573-3587.e29 (2021).

# Session Info

```{r}
sessionInfo()
```